_resources/port1.0/group/haskell_cabal-1.0.tcl

```diff
@@ -32,6 +32,11 @@ proc haskell_cabal.add_dependencies {} {
         depends_build-append \
             port:cabal \
             port:ghc
+        depends_lib-append \
+            port:gmp \
+            port:libiconv \
+            port:ncurses \
+            port:zlib
```

Comment:
It can't be right to add all of these to every single port that uses cabal, can it? `shellcheck` for example only used gmp and libiconv.

Reply:
It's right from the *[Caedite eos](https://en.wikipedia.org/wiki/Caedite_eos._Novit_enim_Dominus_qui_sunt_eius.)* school of coding.

Okay, `cabal` builds all almost certainly will link to `libiconv` and `libgmp`. Some (`adblocktoprivoxy`) link to `zlib`. Some could link to `libncurses`. Arguably it's simpler just to include all these small libraries rather then making special cases every time.
```sh
port contents ghc | egrep -e '\.dylib' | xargs otool -L | egrep -E -e '^[[:space:]]+/opt/local' | sort -u
	/opt/local/lib/libgmp.10.dylib (compatibility version 15.0.0, current version 15.1.0)
	/opt/local/lib/libiconv.2.dylib (compatibility version 9.0.0, current version 9.1.0)
	/opt/local/lib/libncurses.6.dylib (compatibility version 6.0.0, current version 6.0.0)
otool -L `which cabal` | egrep -E -e '^[[:space:]]+/opt/local' | sort -u
	/opt/local/lib/libgmp.10.dylib (compatibility version 15.0.0, current version 15.1.0)
	/opt/local/lib/libiconv.2.dylib (compatibility version 9.0.0, current version 9.1.0)
	/opt/local/lib/libz.1.dylib (compatibility version 1.0.0, current version 1.2.12)
```

But if you prefer special cases, I'll issue a modification.

Reply:
Other ports specify their dependencies individually, and I don't really see any reason why these should be an exception.